src/main/java/org/killbill/billing/plugin/vertex/VertexInvoicePluginApi.java

```diff
@@ -83,6 +97,40 @@ public List<InvoiceItem> getAdditionalInvoiceItems(final Invoice invoice,
         }
     }
 
+    @Override
+    public OnSuccessInvoiceResult onSuccessCall(final InvoiceContext context, final Iterable<PluginProperty> properties) {
+        final String invoiceOperation = PluginProperties.findPluginPropertyValue(INVOICE_OPERATION, properties);
+        if (invoiceOperation == null) {
+            return super.onSuccessCall(context, properties);
+        }
+
+        final Collection<String> docCodes = new HashSet<>();
+        try {
+            // Find existing transactions
+            final List<VertexResponsesRecord> responses = dao.getSuccessfulResponses(context.getInvoice().getId(), context.getTenantId());
+            for (final VertexResponsesRecord response : responses) {
+                docCodes.add(response.getDocCode());
+            }
+        } catch (final SQLException e) {
+            logger.warn("Unable to {} transaction in Vertex", invoiceOperation, e);
+            // Don't fail the whole operation though
+        }
+
+        final VertexApiClient vertexApiClient = vertexApiConfigurationHandler.getConfigurable(context.getTenantId());
+        for (final String docCode : docCodes) {
+            try {
+                if ("void".equals(invoiceOperation)) {
+                    vertexApiClient.deleteTransaction(docCode);
+                }
+            } catch (final ApiException e) {
+                logger.warn("Unable to {} transaction in Vertex", invoiceOperation, e);
+                // Don't fail the whole operation though
+            }
+        }
+
+        return super.onSuccessCall(context, properties);
+    }
+
```

Comment:
we need some tests for this implementation.

Reply:
covered getAdditionalInvoiceItems and onSuccessCall in VertexInvoicePluginApiUnitTest